drv/grapefruit-seq-server/src/main.rs

```diff
@@ -96,59 +63,23 @@ fn main() -> ! {
         }
     }
 
-    match ServerImpl::init(&sys, spi, aux) {
-        // Set up everything nicely, time to start serving incoming messages.
-        Ok(mut server) => {
-            let mut buffer = [0; idl::INCOMING_SIZE];
-            loop {
-                idol_runtime::dispatch(&mut buffer, &mut server);
-            }
-        }
-
-        // Initializing the sequencer failed.
-        Err(e) => {
-            // Tell everyone that something's broken, as loudly as possible.
-            ringbuf_entry!(Trace::StartFailed(e));
-            // Leave FAULT_PIN_L low (which is done at the start of init)
-
-            // All these moments will be lost in time, like tears in rain...
-            // Time to die.
-            loop {
-                // Sleeping with all bits in the notification mask clear means
-                // we should never be notified --- and if one never wakes up,
-                // the difference between sleeping and dying seems kind of
-                // irrelevant. But, `rustc` doesn't realize that this should
-                // never return, we'll stick it in a `loop` anyway so the main
-                // function can return `!`
-                sys_recv_notification(0);
-            }
-        }
+    let mut server = ServerImpl::init(&sys);
+    let mut buffer = [0; idl::INCOMING_SIZE];
+    loop {
+        idol_runtime::dispatch(&mut buffer, &mut server);
     }
 }
 
 #[allow(unused)]
-struct ServerImpl<S: SpiServer> {
+struct ServerImpl {
     jefe: Jefe,
-    sys: sys_api::Sys,
-    seq: SpiDevice<S>,
 }
 
-const FAULT_PIN_L: sys_api::PinSet = sys_api::Port::A.pin(15);
-
-const FPGA_PROGRAM_L: sys_api::PinSet = sys_api::Port::B.pin(6);
-const FPGA_INIT_L: sys_api::PinSet = sys_api::Port::B.pin(5);
-const FPGA_CONFIG_DONE: sys_api::PinSet = sys_api::Port::B.pin(4);
-
-const FPGA_LOGIC_RESET_L: sys_api::PinSet = sys_api::Port::I.pin(15);
-
-impl<S: SpiServer + Clone> ServerImpl<S> {
-    fn init(
-        sys: &sys_api::Sys,
-        spi: S,
-        aux: drv_auxflash_api::AuxFlash,
-    ) -> Result<Self, SeqError> {
+impl ServerImpl {
+    fn init(sys: &sys_api::Sys) -> Self {
         // Ensure the SP fault pin is configured as an open-drain output, and
         // pull it low to make the sequencer restart externally visible.
+        const FAULT_PIN_L: sys_api::PinSet = sys_api::Port::A.pin(15);
```

Comment:
Pins can be named in  app.toml files which helps port tasks to different hardware. e.g. app/gimletlet/app.toml
```
[config.sprot]
pins = [
    # ROT_IRQ (af=15 for EXTI)
    { name = "ROT_IRQ", pin = { port = "D", pin = 0, af = 15, direction = "input"}},
    # SPI6 CS repurposed for debugging
    { name = "DEBUG", pin = { port = "G", pin = 8, af = 0, direction = "output"}}
]
```

Reply:
I see that the only use of this on STM32 is in my code. I still encourage use of the feature but changing it could wait for some cleanup round.

Reply:
Switched to `task_config!` in ec67ba343e689f8a535fb166084247f6be2d83a8